staticaddr/protocol_version.go

```diff
@@ -0,0 +1,63 @@
+package staticaddr
+
+import (
+	"math"
+
+	looprpc "github.com/lightninglabs/loop/swapserverrpc"
+)
+
+// AddressProtocolVersion represents the protocol version (declared on rpc
+// level) that the client declared to us.
+type AddressProtocolVersion uint32
+
+const (
+	// ProtocolVersionLegacy indicates that the client is a legacy version
+	// that did not report its protocol version.
+	ProtocolVersionLegacy AddressProtocolVersion = 0
+
+	// ProtocolVersionUnrecorded is set for swaps were created before we
+	// started saving protocol version with swaps.
+	ProtocolVersionUnrecorded AddressProtocolVersion = math.MaxUint32
+
+	// stableRPCProtocolVersion defines the current stable RPC protocol
+	// version.
+	stableRPCProtocolVersion = looprpc.StaticAddressProtocolVersion_LEGACY_ADDRESS
+)
+
+var (
+	// currentRPCProtocolVersion holds the version of the RPC protocol
+	// that the client selected to use for new swaps. Shouldn't be lower
+	// than the previous protocol version.
+	currentRPCProtocolVersion = stableRPCProtocolVersion
+)
+
+// CurrentRPCProtocolVersion returns the RPC protocol version selected to be
+// used for new swaps.
+func CurrentRPCProtocolVersion() looprpc.StaticAddressProtocolVersion {
+	return currentRPCProtocolVersion
+}
+
+// ResetCurrentProtocolVersion resets the current protocol version to the stable
+// protocol. Note: used in integration tests only!
+func ResetCurrentProtocolVersion() {
```

Comment:
Don't think we even need this, this is coming from the loop protocol version code to aid testing of experimental features.